r/src/compute-exec.cpp

```diff
@@ -146,20 +146,12 @@ std::shared_ptr<compute::ExecNode> ExecNode_Scan(
 
   options->dataset_schema = dataset->schema();
 
-  // ScanNode needs the filter to do predicate pushdown and skip partitions
-  options->filter = ValueOrStop(filter->Bind(*dataset->schema()));
-
   // ScanNode needs to know which fields to materialize (and which are unnecessary)
   std::vector<compute::Expression> exprs;
   for (const auto& name : materialized_field_names) {
     exprs.push_back(compute::field_ref(name));
   }
 
-  options->projection =
-      ValueOrStop(call("make_struct", std::move(exprs),
-                       compute::MakeStructOptions{std::move(materialized_field_names)})
-                      .Bind(*dataset->schema()));
```

Comment:
Again, wasn't sure if this whole chunk should go, but tried removing `.Bind` at the end and then both `.Bind` and `ValueOrStop`, and getting compilation errors.

Reply:
```
options->projection = call("make_struct", std::move(exprs), compute::MakeStructOptions{std::move(materialized_field_names)});
```
?

Reply:
Thanks, appears to, yeah!

Reply:
The removal of this `Bind` seems to be the culprit of the big degradation in both memory and compute times in my use case. ARROW-17559